plugins/out_stackdriver/stackdriver.c

```diff
@@ -726,8 +728,44 @@ static int get_stream(msgpack_object_map map)
     return STREAM_UNKNOWN;
 }
 
-static int pack_json_payload(int operation_extracted, int operation_extra_size, 
-                             msgpack_packer* mp_pck, msgpack_object *obj)
+static void validate_insert_id(msgpack_object * subobj, const msgpack_object * o, 
+                               insert_id_status * status)
+{
+    int i = 0;
+    msgpack_object_kv * p = NULL;
+    *status = INSERTID_NOT_EXISTED;
+
+    if (o == NULL || subobj == NULL) {
+        return;
+    }
+
+    for (i = 0; i < o->via.map.size; i++) {
+        p = &o->via.map.ptr[i];
+        if (p->key.type != MSGPACK_OBJECT_STR) {
+            continue;
+        }
+        flb_sds_t key_name = flb_sds_create_len(p->key.via.str.ptr, p->key.via.str.size);
```

Comment:
this memory allocation is not necessary. If the goal is to compare it against ```INSERTID_IN_JSON``` you can do something like:

```C
if (sizeof(INSERTID_IN_JSON) - 1 == p->key.via.str.size && 
    strncmp(INSERTID_IN_JSON, p->key.via.str.ptr, p->key.via.str.size) == 0) {
    ...
    ...
}
```

Reply:
done